### ISSUE TITLE
Processing Long Audioclips

### ISSUE THREAD

**Issue by GoroDendrin** [Macoron/whisper.unity#72]
Currently, in the documentation of ChangeSampleRate in the AudioUtils, it's noted that it currently causes an issue with really long audio Inputs.

I was wondering what, specifically, the root issue is, if it's known already? I'm probably gonna end up hacking together some sort of a solution for this problem for my own project, but it'd sure be nice to know where to start at least.

**Comment by Macoron** [comment]
Not sure what you are referring to. I can't find such comment in `ChangeSampleRate` function.

https://github.com/Macoron/whisper.unity/blob/25c8d267cf219f6a6728c6b35d6566c4a4d1aefd/Packages/com.whisper.unity/Runtime/Utils/AudioUtils.cs#L43-L66

**Comment by GoroDendrin** [comment]
Ah, so it seems. That must have been my own comment. Whoops.

Regardless, I've made sure that my working code matches the project, and the issue persists.

When attempting to process a ~28minute .mp3 file, I end up with an 'out of bounds' error at:
                    dst[i] = Mathf.Lerp(src[low], src[low + 1], dif);

I've tested this with a few other files (All MP3 files, it should be noted), that length or longer, and it still has an issue. Not sure what the lower bound of the issue is, but it seems consistent on my end.

Might be me and my hardware, or it might not be. Either way, it might be worth a look.
